thunder/executors/nvfuserex_impl.py

```diff
@@ -325,7 +374,33 @@ def translate_bound_symbol(bsym: BoundSymbol) -> Any:
             nvout = lc_to_nv_map[out]
             fd.add_output(nvout)
 
-    return fd
+    bind(fd, definition, "definition")
```

Comment:
Instead of binding, can we do the following? 

```
def define_math(fd):
  ...

if (any input is DTensor) {
  class Model(FusionDefinition):
    def __init__(self):
      super().__init__(max_length=...)

    def definition(self):
      define_math(self)

    def multidevice_schedule(self):
      ...

  return Model()
} else {
  fd = FusionDefinition(max_length=...)
  ...
  define_math(fd);
  return fd;
}
```

This is essentially what I wrote at https://github.com/NVIDIA/Fuser/blob/9fbfa14141603ab908791a8af0f074e228d95cbe/tests/python/multidevice/fusion_definition_wrapper.py#L38